UnityProject/Assets/Scripts/Player/PlayerSprites.cs

```diff
@@ -32,9 +29,6 @@ protected void Awake()
 		{
 			clothes[c.name] = c;
 		}
-		playerScript = GetComponent<PlayerScript>();
-		playerScript.OnCharacterSettingsChange.AddListener(OnCharacterSettingsChange);
-		OnCharacterSettingsChange(playerScript.CharacterSettings);
```

Comment:
I don't like this removal of the UnityEvent. This event hook was here for inversion of control, allowing any component to listen for character setting changes. Now this is forcing the SpawnHandler to know about and call each component rather than letting each component subscribe to the event. You are also exposing OnCharacterSettingsChange as a public method now, allowing anything to invoke it. This change removes the inversion of control and more tightly couples PlayerSprites with SpawnHandler.

Reply:
Though on second thought maybe it needed to be this way in order to get around issues with the order in which the Awake methods are called?

Reply:
Yeah nvm, untiy events makes it needlessly complicated.